### PR TITLE
make VideoClips use O(1) rather than O(N) PT tensors

### PR DESCRIPTION
**Pull request by stephenyan1231**
Summary:
- PT dataloader forks worker process to speed up the fetching of dataset example.  The recommended way of multiprocess context is `forkserver` rather than `fork`.

- Main process and worker processes will share the dataset class instance, which avoid duplicating the dataset and save memory. In this process, `ForkPickler(..).dumps(...)` will be called to serialize the objects, including objects within dataset instance recursively. `VideoClips` instance internally uses O(N) `torch.Tensor` to store per-video information, such as pts, and possible clips, where N is the No. of videos.

- During dumping, each `torch.Tensor` will use one File Descriptor (FD). The OS default max limit of FD is 65K by using `ulimit -n` to query. The number of tensors in `VideoClips` often exceeds the limit.

- To resolve this issue, we use a few big tensors in `VideoClips` by concatenating small tensors. This will only require O(1) tensors.

- This is a BC-breaking change in the sense the format of metadata of `VideoClips` is changed, and needs to be re-computed.

Differential Revision: D19173248

### CODE REVIEW

**Comment by bjuncek** [comment]
Hi @stephenyan1231, did you by any chance test out the benefits (in terms of speed of dataset loading) and verified that the accuracy after training remained unchanged?

**Comment by fmassa** [comment]
Subsumed by https://github.com/pytorch/vision/pull/1795
